Sources/Logging/Logging.swift

```diff
@@ -166,6 +166,20 @@ extension Logger {
         }
     }
 
+    /// Convenience function for merging multiple metadata items into a Logger's existing metadata.
+    ///
+    /// - parameters:
+    ///    - metadata: The metadata to merge into a Logger's existing metadata.
+    ///
+    /// - note: Logging metadata behaves as a value that means a change to the logging metadata will only affect the
+    ///         very `Logger` it was changed on.
+    @inlinable
+    public mutating func mergeMetadata(_ metadata: Logger.Metadata) {
+        metadata.forEach { (key, value) in
+            self.handler[metadataKey: key] = value
+        }
```

Comment:
I suspect in many (all?) handler implementations this will acquire and release a lock for each value we update. Can we call `merge` directly on the `handler.metadata` to avoid this? i.e. `self.handler.metadata.merge(metadata)`